src/e2e-node/acp.test.ts

```diff
@@ -39,201 +38,260 @@ import {
+    // Trim `https://` from the start of these URLs,
+    // so that GitHub Actions doesn't replace them with *** in the logs.
+    process.env.E2E_TEST_DEV_NEXT_POD!.replace(/^https:\/\//, ""),
+    process.env.E2E_TEST_DEV_NEXT_IDP_URL!.replace(/^https:\/\//, ""),
+    process.env.E2E_TEST_DEV_NEXT_CLIENT_ID!,
+    process.env.E2E_TEST_DEV_NEXT_CLIENT_SECRET!,
+  ],
+  // inrupt.net
+  // Unfortunately we cannot authenticate against Node Solid Server yet, due to this issue:
+  // https://github.com/solid/node-solid-server/issues/1533
+  // Once that is fixed, credentials can be added here, and the other `describe()` can be removed.
+];
 
-    // TODO: Find out why the unauthenticated fetch still fails after setting public access
-    // Answer: Unauthenticated fetch is currently not supported in ESS 1.2
-    await expect(getSolidDataset(sessionDataset)).resolves.toEqual(
-      expect.objectContaining({ graphs: { default: {} } })
-    );
-  });
-
-  it("can get and set full access for an agent", async () => {
-    const agent = "https://example.org/alice";
-    const agentAccess = await setAgentAccess(
-      sessionDataset,
-      agent,
-      {
-        read: true,
-        append: true,
-        write: true,
-        controlRead: true,
-        controlWrite: true,
-      },
-      { fetch: session.fetch }
-    );
-    expect(agentAccess).toStrictEqual({
-      read: true,
-      append: true,
-      write: true,
-      controlRead: true,
-      controlWrite: true,
-    });
-    expect(await getAgentAccess(sessionDataset, agent, options)).toStrictEqual({
-      read: true,
-      append: true,
-      write: true,
-      controlRead: true,
-      controlWrite: true,
-    });
-  });
-
-  it("can remove access for an agent", async () => {
-    const agent = "https://example.org/malika";
-    const agentAccess = await setAgentAccess(
-      sessionDataset,
-      agent,
-      {
-        read: true,
-        append: true,
-        write: true,
-        controlRead: true,
-        controlWrite: true,
-      },
-      { fetch: session.fetch }
-    );
-    const removedAgentAccess = await setAgentAccess(
-      sessionDataset,
-      agent,
-      {
-        read: true,
-        append: true,
-        write: false,
-        controlRead: false,
-        controlWrite: false,
-      },
-      { fetch: session.fetch }
-    );
-    expect(removedAgentAccess).toStrictEqual({
-      read: true,
-      append: true,
-      write: false,
-      controlRead: false,
-      controlWrite: false,
-    });
-    expect(await getAgentAccess(sessionDataset, agent, options)).toStrictEqual({
-      read: true,
-      append: true,
-      write: false,
-      controlRead: false,
-      controlWrite: false,
+// End-to-end tests against ESS are temporarily disabled while we work out
+// how to authenticate against it now that refresh tokens are rotated after
+// every request:
+// eslint-disable-next-line jest/no-disabled-tests
+describe.each(serversUnderTest)(
```

Comment:
I think this adds unnecessary complexity, let's actually define a clearer e2e testing strategy; this way of defining a testing matrix inside a testsuite is kinda strange, and means we're running many more tests than we really need to be.